recipes/buildah/meta.yaml

```diff
@@ -0,0 +1,80 @@
+{% set version = "1.17.0" %}
+
+package:
+  name: buildah
+  version: {{ version }}
+
+source:
+  url: https://github.com/containers/buildah/archive/v{{ version }}.tar.gz
+  sha256: 194bb1028e610dab576b04dde0d234f971eaadf38731ec9a85114a170ec59faa
+  patches:
+    # NOTE: This adds the memfd_create syscall which is not available on CentOS 6's kernels.
+    - 0001-Add-missing-defines-for-CentOS-6.patch
+    - 0002-Avoid-LDFLAGS-override.patch
+    - 0003-Use-CPP-CC-and-flags-in-dep-check-scripts.patch
+    - 0004-Use-config-files-in-prefix-as-fallback.patch
```

Comment:
I intend to
- not upstream `0001-Add-missing-defines-for-CentOS-6.patch`.
  IMO, we (conda-forge) should update to cos7 in the foreseeable future.
  And I don't want to burden upstream with temporarily adding these patches.
- upstream `0002-Avoid-LDFLAGS-override.patch`.
- upstream `0003-Use-CPP-CC-and-flags-in-dep-check-scripts.patch`.
- work with upstream on how we can support the Conda use case with something less ugly than
  `0004-Use-config-files-in-prefix-as-fallback.patch`.
  (Upstream was very forthcoming when @epruesse packaged `skopeo`, hence I believe we can work this out.)
  The patch works and I intend to maintain it until we have upstream support, but it's quite hacky and ideally we'd
  have something that doesn't involve hard-coding prefix paths etc.)